crates/turbopack-ecmascript/src/tree_shake/asset.rs

```diff
@@ -52,20 +49,27 @@ impl Asset for EcmascriptModulePartAsset {
     fn content(&self) -> AssetContentVc {
         // This is not reachable because EcmascriptModulePartAsset implements
         // ChunkableAsset and ChunkableAsset::as_chunk is called instead.
-        unreachable!("EcmascriptModulePartAsset::content cannot be called directly")
+        todo!("EcmascriptModulePartAsset::content is not implemented")
     }
 
     #[turbo_tasks::function]
     async fn references(&self) -> Result<AssetReferencesVc> {
         let split_data = split_module(self.full_module).await?;
         let part_id = match get_part_id(&split_data, self.part).await {
             Ok(v) => v,
-            Err(_) => return Ok(self.full_module.references()),
+            Err(_) => bail!("part {:?} is not found in the module", self.part),
         };
 
-        let deps = match split_data.deps.get(&part_id) {
+        let deps = match &*split_data {
+            SplitResult::Ok { deps, modules, .. } => deps,
+            _ => {
+                bail!("failed to split module")
+            }
+        };
```

Comment:
Swap that with the previous statement to avoid calling `get_part_id` with a non Ok SplitResult.